### PR TITLE
feat(devops): Deploy signer from pre-built release

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The signer is now hosted in a separate repository and is available as a pre-built Wasm

# Changes
* Move the signer canister ids from `canister_ids.json` to `dfx.json` as befits an external canister.
* Simplify:
  * The signer now has only signing APIs, so there are fewer bindings.
  * The deploy args are simplified:
    * Some fields are no longer in the deploy args.
    * We can _probably_ reduce the ECDSA keys to 1
  * Deploying is now just `dfx deploy`

# Tests
Existing CI  should suffice

### CODE REVIEW

File: .nvmrc
Comment (lmuntaner) [inline .nvmrc:1]: Could we add this in another PR?

I think they prefer that each PR has one single topic.

File: dfx.json [hunk 1]
```diff
@@ -2,11 +2,19 @@
...
 		"signer": {
-			"candid": "src/backend/backend.did",
-			"package": "backend",
-			"type": "rust",
-			"optimize": "cycles",
-			"gzip": true
+			"type": "custom",
+			"candid": "https://raw.githubusercontent.com/dfinity/chain-fusion-signer/v0.1.1/src/signer/signer.did",
+			"wasm": "https://github.com/dfinity/chain-fusion-signer/releases/download/v0.1.1/signer.wasm.gz",
+			"init_arg": "(variant { Init = record { ecdsa_key_name = \"key_1\"; ic_root_key_der = null;}})",
```
Comment (lmuntaner) [inline dfx.json:8]: `"key_1"` is only for local development, right? And is `ic_root_key_der` different for local?

File: src/declarations/signer/signer.did
Comment (lmuntaner) [inline src/declarations/signer/signer.did:1]: Is this copied over with `dfx generate`?

**Review by lmuntaner — APPROVED** [review]
Thanks!!!

Just a couple of comments, but already approving.

File: dfx.json (hunk 1 above)
Comment (bitdivine) [reply dfx.json:8]: key_1 is the prod value.  I was hoping that we could use only prod values here.  I think it's unclear whether that would work so I'll update the PR to use the same keys as before.

File: .nvmrc
Comment (bitdivine) [reply .nvmrc:1]: Good catch.  Committed accidentally.

File: dfx.json (hunk 1 above)
Comment (bitdivine) [reply dfx.json]: Done.  Please see `scripts/build.signer.sh`

File: src/declarations/signer/signer.did
Comment (bitdivine) [reply src/declarations/signer/signer.did:1]: I guess so.  Let me check.  It is created by `npm run generate` which is `./scripts/genmerate.sh`.  If I delete the generated files and regenerate, it is created:
```
$ rm -fr src/declarations/signer/
$ mkdir src/declarations/signer/
$ scripts/generate.sh
$ ls src/declarations/signer/signer.did
src/declarations/signer/signer.did
```

Let's check whether just `dfx generate signer` will also create it:
```
$ rm -fr src/declarations/signer
$ dfx generate signer
$ ls src/declarations/signer/
index.d.ts       index.js         signer.did       signer.did.d.ts  signer.did.js    
$ ls src/declarations/signer/signer.did
src/declarations/signer/signer.did
```
So, yes, but the `npm` command then modifies the files created by `dfx generate`.

File: src/declarations/signer/signer.did [hunk 2]
```diff
@@ -124,66 +42,14 @@ type SignRequest = record {
...
-  chain_id : nat64;
-  contract_address : text;
-  symbol : opt text;
-};
-type UserTokenId = record { chain_id : nat64; contract_address : text };
 service : (Arg) -> {
-  add_user_credential : (AddUserCredentialRequest) -> (Result);
-  bulk_up : (blob) -> ();
   caller_btc_address : (BitcoinNetwork) -> (text);
   caller_eth_address : () -> (text);
```
Comment (lmuntaner) [inline src/declarations/signer/signer.did:47]: Why isn't this did file include `caller_btc_balance`? When did you make the release?

Comment (lmuntaner) [reply src/declarations/signer/signer.did:47]: I saw that the release used is from 2 weeks ago. That answers my question, sorry for the noise!

Comment (bitdivine) [reply src/declarations/signer/signer.did:47]: Two weeks ago.  I can make a new release if wanted?

File: src/declarations/signer/signer.did
Comment (bitdivine) [reply src/declarations/signer/signer.did:1]: The generation is actually really annoying.  It's very tempting to do just 'dfx deploy && dfx generate'  But in another PR.

File: src/declarations/signer/signer.did (hunk 2 above)
Comment (lmuntaner) [reply src/declarations/signer/signer.did:47]: Yes, but not for this PR. Let's get this in first.

File: src/declarations/signer/signer.did
Comment (lmuntaner) [reply src/declarations/signer/signer.did:1]: I can believe that...

File: src/declarations/signer/signer.did (hunk 2 above)
Comment (lmuntaner) [reply src/declarations/signer/signer.did:47]: Great thanks!

**Pull request merged**